### PR TITLE
Mepts | 1061 Changes MI Cat 15

### PR DESCRIPTION
**Pull request by eliazmuchave**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 1]
```diff
@@ -1455,16 +1453,7 @@ public CohortDefinition getMI15G() {
...
-            + "           Max(e.encounter_datetime) AS last_consultation_date "
-            + "                                     FROM   encounter e "
-            + "                                     WHERE  e.voided = 0 "
-            + "                                            AND e.location_id = :location "
-            + "                                            AND e.encounter_type = ${6} "
-            + "                                            AND e.patient_id = p.patient_id "
-            + "                                            AND e.encounter_datetime BETWEEN "
-            + "                                                :startDate AND :endDate "
-            + "                                     LIMIT  1)  ";
+            + "       AND ee.encounter_datetime <  :endDate ";
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java]:
```suggestion
            + "       AND ee.encounter_datetime <=  :endDate ";
```

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 2]
```diff
@@ -1656,30 +1645,33 @@ public CohortDefinition getMI15F() {
...
+            + "       AND NOT EXISTS (SELECT e.encounter_id "
+            + "                       FROM   encounter e    "
+            + "                              INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + "                       WHERE  e.encounter_type = ${6} "
+            + "                              AND e.location_id = :location "
+            + "                              AND o.concept_id IN( ${856}, ${1305} ) "
+            + "                              AND e.patient_id = p.patient_id "
+            + "                              AND e.voided = 0 "
+            + "                              AND o.voided = 0)"
+            + " GROUP BY p.patient_id         ";
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java]: Spec

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 3]
```diff
@@ -9500,6 +9534,74 @@ public CohortDefinition getPatientsWithDispensationBeforeLastConsultationDate(
...
+            + "INNER JOIN ( "
+            + "                      SELECT     p.patient_id, MAX(e.encounter_datetime) consultation_date "
+            + "                      FROM       patient p "
+            + "                      INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "                      WHERE      e.encounter_type = ${6} "
+            + "                      AND        e.location_id = :location "
+            + "                      AND        e.encounter_datetime BETWEEN :startDate AND :endDate "
+            + "                      AND        p.voided = 0 "
+            + "                      AND        e.voided = 0 "
+            + "                      GROUP BY   p.patient_id ) consultation "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Falta ON

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
Please have a look

**Comment by ningosi** [comment]
@eliazmuchave there is a change request in this PR that @psmatsinhe requested. He needs to review against what he requested and approve that before anyone else can act on it

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java [hunk 4]
```diff
@@ -1253,7 +1253,7 @@ public DataSetDefinition constructIntensiveMonitoringDataSet() {
             "MI15DEN16",
             EptsReportUtils.map(
                 qualityImprovement2020CohortQueries.getMQDen15Dot16(),
-                "startDate=${revisionEndDate-2m+1d},endDate=${revisionEndDate-1m},location=${location}"));
+                "startDate=${startDate},endDate=${endDate},location=${location}"));
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java:1256]: I think this line should be

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java:1256]: startDate=${startDate},
endDate=${revisionEndDate},
location=${location}

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java [hunk 5]
```diff
@@ -1269,7 +1269,7 @@ public DataSetDefinition constructIntensiveMonitoringDataSet() {
             "MI15NUM16",
             EptsReportUtils.map(
                 qualityImprovement2020CohortQueries.getMQNum15Dot16(),
-                "startDate=${revisionEndDate-2m+1d},endDate=${revisionEndDate-1m},location=${location}"));
+                "startDate=${startDate},endDate=${endDate},location=${location}"));
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java:1272]: same here

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/IntensiveMonitoringDataSet.java:1272]: startDate=${startDate},
endDate=${revisionEndDate},
location=${location}

**Review by ningosi — COMMENTED** [review]
See my commends on the dates you supplied

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**